src/Take.Blip.Builder/FlowManager.cs

```diff
@@ -501,12 +501,16 @@ await output.Conditions.EvaluateConditionsAsync(lazyInput, context, cancellation
                             if (IsContextVariable(replacedVariable))
                             {
                                 replacedVariable = await _variableReplacer.ReplaceAsync(replacedVariable, context, cancellationToken);
-                                if (replacedVariable.IsNullOrEmpty())
-                                {
-                                    continue;
-                                }
                             }
-                            state = flow.States.SingleOrDefault(s => s.Id == replacedVariable);
+                            state = flow.States.FirstOrDefault(s => s.Id == replacedVariable);
+
+                            if (state == null)
+                            {
+                                await _stateManager.DeleteStateIdAsync(context, cancellationToken);
+
+                                throw new ArgumentException($"Failed to process output condition, bacause the output context variable '{output.StateId}' is undefined or does not exist in the context.");
```

Comment:
I think `InvalidOperationException` is a better option here. `ArgumentException` usually is used when method's parameters are invalid.

Reply:
Fixed.